sdk/core/src/policies/retry_policies/retry_policy.rs

```diff
@@ -13,13 +14,18 @@ use std::time::Duration;
 /// All retry policies follow a similar pattern only differing in how
```

Comment:
This comment is not really correct today (and points to the fact that this trait wasn't originally intended to be extended by end users). Can we update the comment to reflect the new reality?